core/src/main/java/tc/oc/pgm/kits/Slot.java

```diff
@@ -116,8 +128,8 @@ protected Range<Integer> getAutoIndexRange() {
      * Convert a {@link PlayerInventory} slot index to a {@link Slot} object. Returns null if the
      * index is out of range.
      */
-    public static @Nullable Slot forIndex(int index) {
-      return byIndex.get(index);
+    public static @Nullable Player forIndex(int index) {
+      return (Player) byIndex.get(Player.class, index);
     }
```

Comment:
Why is the return type of slot relevant to the caller? i think these classes should essentially be irrelevant to the caller "as long as it works when using the slot" which, it should

```suggestion
    public static @Nullable Slot forIndex(int index) {
      return byIndex.get(Player.class, index);
    }
```

Reply:
Ah yes, this makes more sense when the inventory holder types were different.